### PR TITLE
User forms branch

### PR DESCRIPTION
**Pull request by Oitur**
Finished project, submitting for review

### CODE REVIEW

**Comment by Oitur** [comment]
We have been trying (unsuccessfully) to replicate the error, however it seems that the problem you had was a local one, as we've ran the bundle installation on several systems. I'll request for a new review to see if this is something that happens elsewhere (perhaps to another reviewer) and try to fix it if we can replicate.

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.update(params[:id], user_params)
+    redirect_to edit_user_path
+  end
+
```
Comment (ebukaume) [inline app/controllers/users_controller.rb]: You are expected to consider situations where the edit could not be saved.

* [ ] `redirect_to edit_user_path` if user was saved
* [ ] `render :edit` if user was not saved ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_update.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))(https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_update.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    # @user = User.new(username: params[:username], email: params[:email], password: params[:password])
```
Comment (ebukaume) [inline app/controllers/users_controller.rb]: Please remove this commented code. Try not to submit projects with commented code unless you are asked to do so.

- [ ] If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

File: app/views/users/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,38 @@
+ <%
+=begin%>
+ <form action="/users" method="post" accept-charset="UTF-8">
```
Comment (ebukaume) [inline app/views/users/new.html.erb]: You should get used to using the named routes provided by rails. In this case: `new_user_path` in place of `/users`.

File: app/views/users/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  <input type="text" name="user[username]"><br>
+  <label for="email"> Email: </label>
+  <input type="email" name="user[email]"><br>
+  <label for="password"> Password: </label>
+  <input type="password" name="user[password]"><br>
+  <label for="password_confirmation"> Confirmation: </label>
+  <input type="password" name="user[password_confirmation]">
+  <input type="submit" value="Submit">
+  <input type="hidden" name="authenticity_token" value="<%= form_authenticity_token %>">
+</form>
```
Comment (ebukaume) [inline app/views/users/new.html.erb]: You also seem to have omitted the labels. How would the user know the data to input? Please fix that too.

File: app/views/users/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+<%
+=begin%>
+ <%= form_tag('/users', method: :post) do -%>
+  <div><%= text_field_tag 'username' %></div>
+  <div><%= email_field_tag 'email' %></div>
+  <div><%= password_field_tag 'password' %></div>
+  <div><%= password_field_tag 'password_confirmation' %></div>
+  <div><%= submit_tag 'Save' %></div>
+<% end -%> 
+<%
```
Comment (ebukaume) [inline app/views/users/new.html.erb]: Named routes and labels here too.

File: app/views/users/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+=end%>
+
+<%= form_for @user do |form|%>
+  <%= form.label :username, 'Your username'%>
+  <%= form.text_field :username, placeholder: "Your name here" %>
+  <%= form.email_field :email, value: "foo@bar.com" %>
+  <%= form.password_field :password %>
+  <%= form.password_field :password_confirmation %>
+  <%= form.submit %>
+<% end%>
```
Comment (ebukaume) [inline app/views/users/new.html.erb]: Labels...

File: app/views/users/new.html.erb (hunk 3 above)
Comment (Oitur) [reply app/views/users/new.html.erb]: This part is commented out, we left it there as was required by the excercise.

File: app/views/users/new.html.erb (hunk 4 above)
Comment (Oitur) [reply app/views/users/new.html.erb]: This is also commented as per the instructions, however, we can see that there are no labels on the actual page and will correct them

File: app/views/users/new.html.erb (hunk 5 above)
Comment (Oitur) [reply app/views/users/new.html.erb]: Also a commented part as per the instructions (perhaps we should remove them?)

File: app/views/users/new.html.erb (hunk 6 above)
Comment (Oitur) [reply app/views/users/new.html.erb]: Labels corrected :)

**Pull request merged**